airflow/models/skipmixin.py

```diff
@@ -65,13 +61,44 @@ def skip(self, dag_run, execution_date, tasks, session=None):
                 ti.end_date = now
                 session.merge(ti)
 
-            session.commit()
+    @provide_session
+    def skip(
+        self,
+        dag_run,
+        execution_date,
+        tasks,
+        session=None,
+        ti: Optional[TaskInstance] = None,
+    ):
+        """
+        Sets tasks instances to skipped from the same dag run. If ti is given, store the list of
+        skipped task IDs to XCom so that NotPreviouslySkippedDep knows these tasks should be skipped
+        when they are cleared.
+
+        :param dag_run: the DagRun for which to set the tasks to skipped
+        :param execution_date: execution_date
+        :param tasks: tasks to skip (not task_ids)
+        :param session: db session to use
+        :param ti: The TaskInstance that initiates the skip. Used for storing results to XCom.
+            If not given, skipped task IDs will not be stored to XCom.
+        """
+        if not tasks:
+            return
+
+        self._set_state_to_skipped(dag_run, execution_date, tasks, session)
+        session.commit()
+        if ti:
```

Comment:
We can avoid needing a TI here, and thus not have to change the fn sigature by calling `XCom.set` directly:

```
XCom.set(
    key=XCOM_SKIPMIXIN_KEY,
    value={"skipped": [d.task_id for d in tasks],
    execution_date=execution_date,
    task_id=self.task_id,
    dag_id=dag_run.dag_id,
    session=session
)
```

Reply:
Oh good idea. One thing i realised though is that `SkipMixin` is not necessarily a `BaseOperator`. So it may not have a `self.task_id` attribute. However, I scanned through the code base I think that only happened in tests. All real use of `SkipMixin` are creating `BaseOperator` from it. Nevertheless I added a type check to make sure this does not break existing code/tests and only do XCom.set() if `self.task_id` is valid.